streamer/pipeline_configuration.py

```diff
@@ -242,9 +242,13 @@ class PipelineConfig(configuration.Base):
   multiple inputs.
   """
 
-  # TODO(joeyparrish): Default to whatever is in the input.
-  channels = configuration.Field(int, default=2).cast()
-  """The number of audio channels to encode."""
+  channel_layouts = configuration.Field(
+      List[bitrate_configuration.AudioChannelLayoutName],
+      required=True).cast()
```

Comment:
I don't agree with making this a required field.  I think we should have a default value instead.  Perhaps `['surround', 'stereo']`.  Not sure.  What are your thoughts?

Reply:
I fear that someone might provide a generic bitrate config that doesn't have `stereo` in it.

My thoughts is that we can default to a list that contains only the lowest `channel_layout`. So we read the `channel_layouts` using the `BitrateConfig` class and then provide this list as a default `[AudioChannelLayout.sorted_values()[0].get_key()]`, now it will pick `stereo` by default, or, when a generic user provided bitrate config is there, it will pick the lowest of what ever the user has.

We can do the same for `resolutions` field as well.

Reply:
Too bad this is not gonna work :(,
Because `channel_layouts` and `resolutions` are class variables and are initialized at indexing time, so even before the bitrate config is provided.

Reply:
If you agree to the proposed idea above, we can monkey-patch the `default` of these fields in the `__init__` of the `PipelineConfig`.
like this.
```
self.__class__.resolutions.default=[ # type: ignore
        bitrate_configuration.VideoResolution.sorted_values()[0].get_key()]
self.__class__.channel_layouts.default=[ # type: ignore
        bitrate_configuration.AudioChannelLayout.sorted_values()[0].get_key()]
```
we need the `# type: ignore` now because since we did the `cast()` for these class attributes, mypy will think they are actual `[RuntimeMap]` while they are actually `Field` instances.